libazureinit/src/media.rs

```diff
@@ -69,40 +71,37 @@ pub fn make_temp_directory() -> Result<(), Box<dyn std::error::Error>> {
 
     create_dir_all(file_path)?;
 
-    let metadata = fs::metadata(file_path).unwrap();
+    let metadata = fs::metadata(file_path)?;
     let permissions = metadata.permissions();
     let mut new_permissions = permissions.clone();
     new_permissions.set_mode(0o700);
-    fs::set_permissions(file_path, new_permissions).unwrap();
+    fs::set_permissions(file_path, new_permissions)?;
 
     Ok(())
 }
 
-pub fn read_ovf_env_to_string() -> Result<String, Box<dyn std::error::Error>> {
+pub fn read_ovf_env_to_string() -> Result<String, Error> {
     let file_path = "/run/azure-init/tmp/ovf-env.xml";
-    let mut file = File::open(file_path).expect("Failed to open file");
+    let mut file = File::open(file_path)?;
     let mut contents = String::new();
-    file.read_to_string(&mut contents)
-        .expect("Failed to read file");
+    file.read_to_string(&mut contents)?;
 
     Ok(contents)
 }
 
-pub fn parse_ovf_env(
-    ovf_body: &str,
-) -> Result<Environment, Box<dyn std::error::Error>> {
+pub fn parse_ovf_env(ovf_body: &str) -> Result<Environment, Error> {
     let environment: Environment = from_str(ovf_body)?;
 
-    if environment
+    if !environment
         .provisioning_section
         .linux_prov_conf_set
         .password
         .is_empty()
     {
-        return Err("Password is empty".into());
+        Err(Error::NonEmptyPassword)
```

Comment:
Elsewhere we were returning an error if it _was_ empty and users of this function would immediately test that the password was empty so this is a behavior change.